shared/reducers/settings.js

```diff
@@ -7,18 +7,73 @@ const initialState: State = {
   notifications: {
     settings: null,
     unsubscribedFromAll: null,
+    allowSave: false,
+    allowEdit: false,
   },
 }
 
-function reducer (state: State = initialState, action: Actions) {
+function reducer (state: State = initialState, action: Actions): State {
   switch (action.type) {
     case CommonConstants.resetStore:
       return {...initialState}
+    case Constants.notificationsToggle:
+      if (!state.notifications.settings || !state.notifications.allowEdit) {
+        return state
+      }
+
+      // No name means the unsubscribe option
+      const name = action.payload.name
+
+      const updateSubscribe = setting => {
+        let subscribed = setting.subscribed
+
+        if (!name) { // clicked unsub all
+          subscribed = false
+        } else if (name === setting.name) { // flip if its the one we're looking for
+          subscribed = !subscribed
+        }
+
+        return {
+          ...setting,
+          subscribed,
+        }
+      }
+
+      return {
+        ...state,
+        notifications: {
+          ...state.notifications,
+          settings: state.notifications.settings.map(updateSubscribe),
+          unsubscribedFromAll: name ? false : !state.notifications.unsubscribedFromAll,
+          allowSave: true,
+        },
+      }
+    case Constants.notificationsSave:
+      return {
+        ...state,
+        notifications: {
+          ...state.notifications,
+          allowEdit: false,
+        },
+      }
+    case Constants.notificationsSaved:
+      return {
+        ...state,
+        notifications: {
+          ...state.notifications,
+          allowSave: false,
```

Comment:
Maybe `allowSave: true` here? It might be confusing to go away from the window for a bit, come back, and see the save button disabled without context.